src/GuzzleRequestAdapter.php

```diff
@@ -110,21 +130,36 @@ private function tryHandleResponse(RequestInformation $requestInfo, ResponseInte
      */
     public function sendAsync(RequestInformation $requestInfo, array $targetCallable, ?array $errorMappings = null): Promise
     {
-        return $this->getHttpResponseMessage($requestInfo)->then(
-            function (ResponseInterface $result) use ($targetCallable, $requestInfo, $errorMappings) {
-                $response = $this->tryHandleResponse($requestInfo, $result, $errorMappings);
-
-                if ($response !== null) {
-                    return $response;
-                }
-                $this->throwFailedResponse($result, $errorMappings);
-                if ($this->is204NoContentResponse($result)) {
-                    return null;
+        $span = $this->startTracingSpan($requestInfo, 'sendAsync');
+        $scope = $span->activate();
+        $currentContext = $span->storeInContext(Context::getCurrent());
+        try {
+
+            $responseMessage = $this->getHttpResponseMessage($requestInfo, '', $span);
+            $finalResponse   = $responseMessage->then(
+                function (ResponseInterface $result) use ($targetCallable, $requestInfo, $errorMappings, &$span) {
+                    $response = $this->tryHandleResponse($requestInfo, $result, $errorMappings);
+                    if ($response !== null) {
+                        $span->addEvent(self::EVENT_RESPONSE_HANDLER_INVOKED_KEY);
+                        return $response;
+                    }
+
+                    $this->throwFailedResponse($result, $errorMappings, $span);
+                    $span->setStatus(StatusCode::STATUS_OK, 'response_handle_success');
+                    if ($this->is204NoContentResponse($result)) {
+                        return null;
+                    }
+                    $rootNode          = $this->getRootParseNode($result, $span);
+                    $this->setResponseType($targetCallable[0], $span);
+                    return $rootNode->getObjectValue($targetCallable);
                 }
-                $rootNode = $this->getRootParseNode($result);
-                return $rootNode->getObjectValue($targetCallable);
-            }
-        );
+            );
+            $span->setStatus(StatusCode::STATUS_OK, 'sendAsync() success');
```

Comment:
This should be the http response status code
```suggestion
            $span->setStatus(StatusCode::STATUS_OK, '200');
```